src/query/functions/tests/it/scalars/testdata/cast.txt

```diff
@@ -3137,3 +3204,75 @@ evaluation (internal):
 +--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
 
+ast            : TRY_CAST(((1, TRUE), 1) AS Tuple(Tuple(INT, INT), INT))
+raw expr       : TRY_CAST(tuple(tuple(1_u8, true), 1_u8) AS ((Int32, Int32), Int32))
+checked expr   : TRY_CAST(tuple<(UInt8, Boolean), UInt8>(tuple<UInt8, Boolean>(1_u8, true), 1_u8) AS ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL)
+optimized expr : ((1_i32, 1_i32), 1_i32)
+output type    : ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL
+output domain  : (({1..=1}, {1..=1}), {1..=1})
+output         : ((1_i32, 1_i32), 1_i32)
+
+
+ast            : TRY_CAST(((1, 'a'), 1) AS Tuple(Tuple(INT, INT NULL), INT))
+raw expr       : TRY_CAST(tuple(tuple(1_u8, "a"), 1_u8) AS ((Int32, Int32 NULL), Int32))
+checked expr   : TRY_CAST(tuple<(UInt8, String), UInt8>(tuple<UInt8, String>(1_u8, "a"), 1_u8) AS ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL)
+optimized expr : ((1_i32, NULL), 1_i32)
+output type    : ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL
+output domain  : (({1..=1}, {NULL}), {1..=1})
+output         : ((1_i32, NULL), 1_i32)
+
+
+ast            : TRY_CAST(((1, 'a'), 1) AS Tuple(Tuple(INT, INT), INT) NULL)
+raw expr       : TRY_CAST(tuple(tuple(1_u8, "a"), 1_u8) AS ((Int32, Int32), Int32) NULL)
+checked expr   : TRY_CAST(tuple<(UInt8, String), UInt8>(tuple<UInt8, String>(1_u8, "a"), 1_u8) AS ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL)
+optimized expr : ((1_i32, NULL), 1_i32)
+output type    : ((Int32 NULL, Int32 NULL) NULL, Int32 NULL) NULL
+output domain  : (({1..=1}, {NULL}), {1..=1})
+output         : ((1_i32, NULL), 1_i32)
```

Comment:
If this should output (NULL) ?

Reply:
@sundy-li PTAL

Reply:
I think the output type should be `((Int32 NULL, Int32 NULL) NULL, Int32 NULL)`, we should forbidden the use `Tuple` or `Array` inside in `Nullable`.

Reply:
If the original type is not matched, so all the nested types should be null.

Reply:
`TRY_CAST(1 AS TUPLE(INT, INT))` has to output `Null`, therefore the output type of `TRY_CATS(xxx AS TUPLE(INT, INT))` is `TUPLE(Int32 NULL, Int32 NULL) NULL`, which has Tuple inside nullable.

Reply:
Nullable nested types are unavoidable in runtime.

Reply:
Ok, so we should allow this in runtime evaluation.

Reply:
> Nullable nested types are unavoidable in runtime.

It's forbidden in DDL but can be available in runtime.

Reply:
There are two types of errors CAST could throw:

1. trying to cast to unsupported type.
2. The cast overflowed.

So there is another solution to this problem: Allowing `TRY_CAST` to throw errors on the first error type.

Reply:
>  Allowing TRY_CAST to throw errors on the first error type.
That could be better.